### ISSUE TITLE
Scripting Session

### ISSUE THREAD

**Issue by smped** [BIG-SA/BASH-Intro-2018#2]
Hey Jimmy,

Just thinking about that Scripting Session. I'm thinking a nice clear description of things like the following might be good

- What are variables?
- `basename` and `dirname`
- `for` loops 
- File permissions

I don't think we'll have covered these by that week, so it might make a good start. Unless Paul will cover for loops in the sed/awk sessions?

**Comment by jimmybgammyknee** [comment]
Yep so far ive covered variables and for loops (i still need to finish). From what ive seen in the previous materials that John sent me they cover permissions, but I can easily cover that and basename stuff as well

**Comment by smped** [comment]
For sure. I've just finished Section2 & was trying to figure which bits I include/exclude. Feedback welcome, but I think it's looking pretty good.

**Comment by twobeers75** [comment]
Morning Boys, sounds as if you two are all over it but if not, let me know how I can help!

**Comment by jimmybgammyknee** [comment]
Is it possible for someone to introduce `wget` or `curl` in their session? It would be great to enable people to use `wget` in the scripting session instead of relying on files that we provide

**Comment by ppswang** [comment]
Ive included them in cheatsheet, but that's still work in progress.

I can introduce them as incidentals to example questions or exercises.

**Comment by smped** [comment]
They're introduced in Session 2:

https://big-sa.github.io/BASH-Intro-2018/notes/2_filters.html#this-session

**Comment by jimmybgammyknee** [comment]
Ok ive done a rough draft of the entire bash scripting session. Might be a bit raw, so if someone has time, are you able to look through it and suggest any additions/subtractions or interesting exercises? Still quite a bit of time left to clean this up nicely

**Comment by twobeers75** [comment]
Well done Jimmy! I'll take a look sometime over the weekend and will get back to you.

Cheers J

**Comment by twobeers75** [comment]
Hey Again,

Sorry, I didn't get to finish this over the weekend! I just looked at the last bits and pushed all my changes. here are some additional notes for you to address as well;

line 43 - bringing down the particle pollution data. the user should be in a newly created folder called "BashWk4" which is empty. "cd ./BASH-Intro-2018/files/" gives an error? This might be intentional and if they learnt anything from the previous weeks they should be able to create these directories.

line 75-76 - in the basic_example_1.sh script. Added a note just in case they used a different download location and modified the cd command to be absolute from home (cd ~/BashWk4/BASH-Intro-2018/files)

line 90 - questions about first script. They might not realise that the output gets saved in 'ADL07p' so could ask that too?

line 148 - user and group settings. Added a little more detail to that sentence for clarification

line 153 - after changing permisions. you state "As a security measure, Linux doesn’t allow you to execute a script from within the same directory so to execute it enter the following: ./scriptname.sh". Not really an issue as it doesn't matter a great deal but you can actually call a script from within the directory as long as that directory is in your $PATH. Otherwise you specifically need to provide relative or absolute path to script which is why './' works.

line 175 - second bash script iteration. 'cd' command is commented out possibly because we expect them to be in the right directory. Maybe should be an absolute path though?

line 196 - variables. slight tweak about the variables 'ME' and 'MESSAGE' they had not been used yet.

line 206 - minor mods

line 237 - welldone.sh. not sure about Mac and Windows but on linux that line should be something like this; echo -e "Hello ${ME}\n\"${MESSAGE}\"\nWell Done!"

line 272 - count_lines.sh. I don't know jack about this data but I think each line in the file is a reading per hour not month? If so the print line "The number of months contained in the data is ..." should be changed.

line 284 - exercise. need to finish that sentence

line 343 - minor mods

line 411 - basic_example_3.sh. fixed variable name (WORKING_DIRECTORY --> $WORKING_DIR)

Everything else worked out of the box and the flow is brilliant! Well done Jimmy!!

Cheers

John

**Comment by jimmybgammyknee** [comment]
Thanks John! Tweaked a few things again, but will look through again later this week

**Comment by twobeers75** [comment]
Too easy Champ! I’ll give it another run through later in the week as well.
